lib/zip_source_zip_new.c

```diff
@@ -43,20 +43,43 @@ zip_source_t *_zip_source_zip_new(zip_t *srcza, zip_uint64_t srcidx, zip_flags_t
     zip_stat_t st;
     zip_file_attributes_t attributes;
     zip_dirent_t *de;
-    bool partial_data, needs_crc, needs_decrypt, needs_decompress;
+    bool partial_data, needs_crc, encrypted, needs_decrypt, compressed, needs_decompress, reopening;
+    zip_flags_t stat_flags;
+    zip_t *source_archive;
+    zip_uint64_t source_index;
```

Comment:
We need to make sure the returned source is invalidated when `srcza` is closed. I'm not sure what the best way to do that is, so please just add a TODO comment so we won't forget.

Reply:
Maybe by calling `_zip_source_set_source_archive` on each layer? But I'll add a TODO.